src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3184,15 +3184,19 @@ internal virtual ImmutableArray<LabelSymbol> Labels
         /// <param name="patternType">Type to search.</param>
         /// <param name="methodName">Method to search for.</param>
         /// <param name="lookupResult">Passed in for reusability.</param>
+        /// <param name="searchExtensionMethods">True if the lookup should consider extension methods as viable candidates</param>
         /// <param name="warningsOnly">True if failures should result in warnings; false if they should result in errors.</param>
         /// <param name="diagnostics">Populated with binding diagnostics.</param>
         /// <returns>The desired method or null.</returns>
         internal MethodSymbol FindPatternMethod(TypeSymbol patternType, string methodName, LookupResult lookupResult,
```

Comment:
I think we'll end up needing to split this out to handle different patterns differently. For instance, the GetEnumerator lookup will be like.

1. Search on type
2. Look for conversion to IEnumerable
3. Look for extension method

while Dispose will be like

1. Look for conversion
2. Look for instance method
3. Look for extension method

I'm not sure we can really share much between these two.